server/tserver/src/main/java/org/apache/accumulo/tserver/util/CreateEmpty.java

```diff
@@ -0,0 +1,200 @@
+import org.apache.accumulo.core.crypto.CryptoUtils;
+import org.apache.accumulo.core.file.FileSKVWriter;
+import org.apache.accumulo.core.file.rfile.RFileOperations;
+import org.apache.accumulo.core.file.rfile.bcfile.Compression;
+import org.apache.accumulo.core.metadata.UnreferencedTabletFile;
+import org.apache.accumulo.core.spi.crypto.CryptoEnvironment;
+import org.apache.accumulo.core.spi.crypto.CryptoService;
+import org.apache.accumulo.core.spi.file.rfile.compression.NoCompression;
+import org.apache.accumulo.server.ServerContext;
+import org.apache.accumulo.server.fs.VolumeManager;
+import org.apache.accumulo.start.spi.KeywordExecutable;
+import org.apache.accumulo.tserver.logger.LogFileKey;
+import org.apache.accumulo.tserver.logger.LogFileValue;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.beust.jcommander.IParameterValidator;
+import com.beust.jcommander.Parameter;
+import com.beust.jcommander.ParameterException;
+import com.google.auto.service.AutoService;
+
+/**
+ * Create an empty RFile for use in recovering from data loss where Accumulo still refers internally
+ * to a path.
+ */
+@AutoService(KeywordExecutable.class)
+public class CreateEmpty implements KeywordExecutable {
+  private static final Logger LOG = LoggerFactory.getLogger(CreateEmpty.class);
+  public static final String RF_EXTENSION = ".rf";
+  public static final String WAL_EXTENSION = ".wal";
+
+  public static class MatchesValidFileExtension implements IParameterValidator {
+    @Override
+    public void validate(String name, String value) throws ParameterException {
+      if (value.endsWith(RF_EXTENSION) || value.endsWith(WAL_EXTENSION)) {
+        return;
+      }
+      throw new ParameterException("File must end with either " + RF_EXTENSION + " or "
+          + WAL_EXTENSION + " and '" + value + "' does not.");
+    }
+  }
+
+  public static class IsSupportedCompressionAlgorithm implements IParameterValidator {
+    @Override
+    public void validate(String name, String value) throws ParameterException {
+      List<String> algorithms = Compression.getSupportedAlgorithms();
+      if (!algorithms.contains(value)) {
+        throw new ParameterException("Compression codec must be one of " + algorithms);
+      }
+    }
+  }
+
+  static class Opts extends ConfigOpts {
+    @Parameter(names = {"-c", "--codec"}, description = "the compression codec to use.",
+        validateWith = IsSupportedCompressionAlgorithm.class)
+    String codec = new NoCompression().getName();
+    @Parameter(
+        description = " <path> { <path> ... } Each path given is a URL."
+            + " Relative paths are resolved according to the default filesystem defined in"
+            + " your Hadoop configuration, which is usually an HDFS instance.",
+        required = true, validateWith = MatchesValidFileExtension.class)
+    List<String> files = new ArrayList<>();
+
+    public enum OutFileType {
+      rfile, wal
```

Comment:
Enum style is generally all upper-case, like constants. Can use case conversion with `.name()` of the enum, if needed.

Also, I suggest RF rather than "rfile", since that's the filename extension and most intuitive shorthand way of specifying the file's type.

```suggestion
      RF, WAL
```

Reply:
changed in 2124f38165